Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,423 @@
+#pragma once
+
+#include "Layer.h"
+#include "IpAddress.h"
+#include "MacAddress.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @class WireGuardLayer
+	 * Represents a WireGuard protocol layer
+	 */
+	class WireGuardLayer : public Layer
+	{
+	public:
+		/**
+		 * WireGuard message types
+		 */
+		enum class WireGuardMessageType
+		{
+			/** Handshake Initiation message */
+			HandshakeInitiation = 1,
+			/** Handshake Response message */
+			HandshakeResponse = 2,
+			/** Cookie Reply message */
+			CookieReply = 3,
+			/** Transport Data message */
+			TransportData = 4
+		};
+
+#pragma pack(push, 1)
+		/**
+		 * @struct wg_common_header
+		 * Represents the common header for all WireGuard message types
+		 */
+		struct wg_common_header
+		{
+			/** Message type field */
+			uint8_t messageType;
+			/** Reserved field (3 bytes) */
+			uint8_t reserved[3];
+		};
+#pragma pack(pop)
+
+		WireGuardLayer()
+		{}
```

Comment:
This is redundant?

Reply:
This c'tor is still there...

Reply:
The subclasses of WireGuardLayer (WireGuardMessages) require a default constructor to be instantiated without initial data. Removing it leads to compilation errors.

Reply:
You're right! Let's move it to the `protected` area so it won't be public?